### PR TITLE
[Routing] fix to allow verbs http annotation @Method("GET|POST")

### PR DESCRIPTION
**Pull request by sergiors**
Inside $route->setMethods() is called implode()
I think this issue solve #312

### CODE REVIEW

File: Routing/AnnotatedRouteControllerLoader.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ protected function configureRoute(Route $route, \ReflectionClass $class, \Reflec
         // requirements (@Method)
         foreach ($this->reader->getMethodAnnotations($method) as $configuration) {
             if ($configuration instanceof Method) {
-                $route->setMethods(implode('|', $configuration->getMethods()));
+                $route->setMethods(explode('|', implode('|', $configuration->getMethods())));
```
Comment (xabbuh) [inline Routing/AnnotatedRouteControllerLoader.php]: What is the point in calling `implode()` at all if we just call `explode()` with its result?

Comment (sergiors) [inline Routing/AnnotatedRouteControllerLoader.php]: it's ugly, i know.. hehhe
OK, `$configuration->getMethods()` may return 
```
[
     "GET",
     "PUT|POST",
]
```

```
implode('|', $methods); \\ => "GET|PUT|POST"
``` 
as `$route-> setMethods()`works with array, I call explode to normalize

Comment (sergiors) [inline Routing/AnnotatedRouteControllerLoader.php]: `explode` needs a string and `$configuration->getMethods()` return just array

Comment (sergiors) [inline Routing/AnnotatedRouteControllerLoader.php]: @xabbuh ping

Comment (xabbuh) [inline Routing/AnnotatedRouteControllerLoader.php]: Ah I see, makes sense then.

Comment (xabbuh) [inline Routing/AnnotatedRouteControllerLoader.php]: Or can we maybe just change the configuration to already return a flat list of methods?

Comment (sergiors) [inline Routing/AnnotatedRouteControllerLoader.php]: Normalize the array in https://github.com/sensiolabs/SensioFrameworkExtraBundle/blob/master/Configuration/Method.php#L46

?

Comment (xabbuh) [inline Routing/AnnotatedRouteControllerLoader.php]: Looks like it's not needed there. So when could `$configuration->getMethods()` actually return what you posted above?

Comment (sergiors) [inline Routing/AnnotatedRouteControllerLoader.php]: Yes, `$configuration->getMethods()` could return
```
[
     "PUT|POST",
]
```

https://github.com/sensiolabs/SensioFrameworkExtraBundle/pull/380/files#diff-f23783624ea580a4daab1fd95f6a31c0R22

Comment (xabbuh) [inline Routing/AnnotatedRouteControllerLoader.php]: In that line it already is a sequence and is https://github.com/sensiolabs/SensioFrameworkExtraBundle/pull/380/files#diff-f23783624ea580a4daab1fd95f6a31c0R30 really a supported use case?

Comment (xabbuh) [inline Routing/AnnotatedRouteControllerLoader.php]: Under which circumstances?

Comment (sergiors) [inline Routing/AnnotatedRouteControllerLoader.php]: https://github.com/sensiolabs/SensioFrameworkExtraBundle/pull/380/files#diff-f23783624ea580a4daab1fd95f6a31c0R22

Comment (xabbuh) [inline Routing/AnnotatedRouteControllerLoader.php]: Sorry, but I cannot reproduce your observations. That example gives me the following output:

```
Array
(
    [0] => GET
    [1] => POST
)
```

Which version of the bundle did you use (I tested it with the latest version, 3.0.11)?

Comment (sergiors) [inline Routing/AnnotatedRouteControllerLoader.php]: @xabbuh the last version..
I cloned the bundle and add https://github.com/sergiors/SensioFrameworkExtraBundleRouteAnnotation/commit/adc01bb0723805d67e9a2730ddafee8f97fa6f5c

run: `composer install` and `phpunit tests/Routing/AnnotatedRouteControllerLoaderTest.php`

the result:
```
PHPUnit 5.0.0 by Sebastian Bergmann and contributors.

..F                                                                 3 / 3 (100%)

Time: 144 ms, Memory: 13.50Mb

There was 1 failure:

1) Sensio\Bundle\FrameworkExtraBundle\Tests\Routing\AnnotatedRouteControllerLoaderTest::testLoad
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    0 => 'GET'
-    1 => 'POST'
+    0 => 'GET|POST'
 )
```

**Comment by apfelbox** [comment]
@sergiors you read the error message from PhpUnit wrong.

It expected to find `["GET", "POST"]` but it found `["GET|POST"]` (not an array with 3 items. Please look at `-` and `+` at the beginning of the line.

**Comment by sergiors** [comment]
@apfelbox do you run https://github.com/sensiolabs/SensioFrameworkExtraBundle/pull/380#discussion_r45276564 ?

**Comment by apfelbox** [comment]
@sergiors my comment was only pointed at your comment

> ```
> [
>     'GET',
>     'POST',
>     'GET|POST'
> ]
> ```

which is a wrong interpretation of the output from phpunit.

I ran your example and `@Method({"GET", "POST"})` returns `array('GET|POST')`, like it is expected to do. Not sure what you are trying to fix.

As far as I can see, the annotation always only returns an array with exactly one element (`"GET"` or `"GET|POST"`). Do you have a case where it does something different?

**Pull request closed without merging**